### PR TITLE
Merge changes for release 0.6.2

### PR DESCRIPTION
**Pull request by greschd**
@roosre please have a look at the changes, and whether this should be `0.6.2` release, or `0.7.0`. The `main` branch was switched to `0.7.dev0` after the previous release, but AFAICT there are no significant changes which would warrant a minor bump - except for the Python support change.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -124,6 +127,6 @@ ignored-classes = [
 ]
 
 [tool.mypy]
-python_version = 3.12
+python_version = 3.10
```
Comment (roosre) [inline pyproject.toml:130]: Is this change on purpose?

**Review by roosre — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (greschd) [reply pyproject.toml:130]: Yes, this was wrong before. Mypy should check against the oldest supported version.
